### PR TITLE
Pull Commands from config.yaml

### PR DESCRIPTION
**Pull request by FrankKerschbaumer3**
- Add Unmarshall key and known command struct
- Add knownCommands range
- Add knownCommands cmd to parse config.yaml based on the known: key

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -81,7 +81,28 @@ func main() {
...
+
+			cmd := bot.Command{
+				Prefix:          cmd.Prefix,
+				MessageTemplate: bot.NewHandlerTemplate(cmdTemplate),
+			}
+
+			commands = append(commands, cmd)
+		}
+
+		fmt.Printf("Bot commands: %v", commands)
```
Comment (MatthewEdge) [inline main.go]: Can we remove this debug statement?

File: config/config.go [hunk 2]
```diff
@@ -105,6 +105,19 @@ func (c *Config) CommandsEnabled() bool {
...
+// KnownCommands returns a slice of map[prefix]message pairs, to be parsed elsewhere,
+// that represent commands the Bot responds to
+type KnownCommand struct {
+	Prefix  string `mapstructure:"prefix"`
+	Message string `mapstructure:"message"`
+}
+
+func (c *Config) KnownCommands() []KnownCommand {
+	var commands []KnownCommand
+	c.config.UnmarshalKey(c.key("commands.known"), &commands)
```
Comment (MatthewEdge) [inline config/config.go:117]: This is _sick_. Nice find!

File: main.go (hunk 1 above)
Comment (FrankKerschbaumer3) [reply main.go]: Done

File: config/config.go (hunk 2 above)
Comment (FrankKerschbaumer3) [reply config/config.go:117]: 💯 🥇

**Review by MatthewEdge — APPROVED** [review]
This is awesome! Thank you!

**Pull request merged**